nemo/lightning/ckpt_utils.py

```diff
@@ -12,18 +11,14 @@ def idempotent_path_append(base_dir: Union[str, Path], suffix) -> Path:
 
     assert isinstance(base_dir, Path)
     if base_dir.parts[-1] != suffix:
-        base_dir = base_dir / suffix
+        maybe_base_dir = base_dir / suffix
+        if maybe_base_dir.is_dir():
```

Comment:
IIUC, this function needs to be stateless but not is stateful on disk contents; if the need here is to support BW compatibility, then the if statement needs to go one level above.

Reply:
Sorry, I don't quite follow. What exactly is the issue with this function being stateful?

Reply:
when loading it works as expected, but when saving the suffix directory might not (most likely does not) exist but we want to enforce saving under the /suffix directory,